File: 0242-valid-anagram/0242-valid-anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        if (s.length() != t.length()) {
            return false;
        }

        std::array<int, 26> count = {0};
        for (int i = 0; i < s.length(); i++) {
            count[s[i] - 'a'] += 1;
            count[t[i] - 'a'] -= 1;
        }

        for (int c: count) {
            if (c) return false;
        }

        return true;
    }
};

File: 0242-valid-anagram/0242-valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False
        char_map = {}
        for c in s: 
            char_map[c] = char_map.get(c, 0) + 1
        for c in t: 
            if c not in char_map or char_map[c] == 0: 
                return False
            char_map[c] -= 1
        return True
        